### DISCUSSION TITLE
Cannot add `Selectable` due to table not being in plural.

### DISCUSSION THREAD

**Discussion by wizardlink** [diesel-rs/diesel#3630] [Q&A] [answered]
I am currently dabbling with `diesel` in hopes to become the primary ORM of an existing application I have, essentially in this application I have a table called `self_role` in which I have defined this struct:

```rs
#[derive(Queryable)]
pub struct SelfRole {
    pub id: i64,
    pub style: i32,
    pub emoji: Option<String>,
}
```

Alright, so inserting works just fine but following the [Getting Started](https://diesel.rs/guides/getting-started) I cannot actually accomplish the `src/bin/show_posts.rs` example, I can call `filter` but it doesn't chains into `limit` and I don't see anything I missed in terms of derivatives/traits.

Upon looking for an answer I stumbled upon the [`Selectable`](https://docs.rs/diesel/latest/diesel/prelude/trait.Selectable.html) trait, however upon using it, whatever I may name my struct, it always appends it with `s` (thus assuming a plural schema). This leads to the error:
```
failed to resolve: use of undeclared crate or module `self_roles`
```

By changing manually the schema to become `self_roles` the error goes away, but like I mentioned in the beginning of this post, this is an existing application and I'd rather not change it's existing structure. Is there any reason for the trait behaving this way, or am I missing something?

**Comment by weiznich** [comment, marked as answer]
> I can call filter but it doesn't chains into limit and I don't see anything I missed in terms of derivatives/traits.

I'm not sure what are you referring to here. Did you try to add `.limit()` and it fails to compile with rustc? In that case it would be helpful to get the compiler error and the relevant code. Or did it not show up in the code completions in your IDE? In that case: The method exists, but rust-analyzer is not able to find it for whatever reason (https://github.com/rust-lang/rust-analyzer/issues/14607)

> Upon looking for an answer I stumbled upon the [Selectable](https://docs.rs/diesel/latest/diesel/prelude/trait.Selectable.html) trait, however upon using it, whatever I may name my struct, it always appends it with s (thus assuming a plural schema). This leads to the error:

That's the default behavior. You can change it as [described in the documentation](https://docs.diesel.rs/2.0.x/diesel/expression/derive.Selectable.html) by adding a `#[diesel(table_name = crate::schema::self_role)]`. Note that you've linked the documentation of the trait which is something different than the derive. The former specifies the general behavior while, the later implements a specific instance.

**Reply by wizardlink** [reply]
Firstly, thank you for taking the time of your day to answer this!

You are correct about `rust-analyzer`, since I didn't see it in completion like other methods I quickly jumped the gun thinking it was incorrect, thank you for pointing it out.

> That's the default behavior. You can change it as [described in the documentation](https://docs.diesel.rs/2.0.x/diesel/expression/derive.Selectable.html) by adding a `#[diesel(table_name = crate::schema::self_role)]`. Note that you've linked the documentation of the trait which is something different than the derive. The former specifies the general behavior while, the later implements a specific instance.

And you're correct again, in the end I just had to specify the desired table name. I started learning Rust not too long ago so these concepts are still a little confusing as I haven't implemented any on my own. Thank you for the clarification, I'll be reading into it!

**Reply by weiznich** [reply]
Thanks for coming back and confirming that it work. And no worries to ask these kind of questions, here is the right place to ask them, although it might help next time to include more context.

**Comment by auroraanna** [comment]
I am having this issue too with relations. If my tables names are singular, even if the table name is specified, the belonging_to() function doesn't work.
